### PR TITLE
Meshing queries refactoring, improved error handling and doc updates

### PR DESCRIPTION
**Pull request by hpohekar**
image

`---------------------------------------------------------------`

image

`---------------------------------------------------------------`

image

`---------------------------------------------------------------`

image

### CODE REVIEW

**Comment by seanpearsonuk** [comment]
@hpohekar this is an observation rather than a criticism. The summary of each of these "service" objects focuses on the implementation: "wrapper over ... gRPC" rather than telling the reader what the "service" can offer them. This is bewildering for te average reader.

File: src/ansys/fluent/core/services/meshing_queries.py [hunk 1]
```diff
@@ -624,7 +624,16 @@ def __init__(self, service: MeshingQueriesService):
     docstring = None
 
     def GetFaceZoneAtLocation(self, location) -> Any:
-        """GetFaceZoneAtLocation."""
+        """
+        Return face zone at or closest to a specified location.
+
+        Note:  This function is not applicable to polyhedra meshes.
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/meshing_queries.py:630]:
```suggestion
        Note:  This function is not applicable to polyhedral meshes.
```

Comment (seanpearsonuk) [reply src/ansys/fluent/core/services/meshing_queries.py:630]: The ansys doc is a typo or some other mistake

Comment (hpohekar) [reply src/ansys/fluent/core/services/meshing_queries.py]: Done. Thank you.

**Comment by hpohekar** [comment]
> @hpohekar this is an observation rather than a criticism. The summary of each of these "service" objects focuses on the implementation: "wrapper over ... gRPC" rather than telling the reader what the "service" can offer them. This is bewildering for te average reader. #1586

Right.

File: src/ansys/fluent/core/services/meshing_queries.py [hunk 2]
```diff
@@ -642,85 +658,170 @@ def GetCellZoneAtLocation(self, location) -> Any:
         return response.cell_zone_id
 
     def GetZonesOfType(self, type) -> Any:
-        """GetZonesOfType."""
+        """
+        Return a list of zones of the specified default zone type.
+
+        .. code-block:: python
+
+            >>> meshing_session.meshing_queries.GetZonesOfType("velocity-inlet")
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/meshing_queries.py:666]: I feel that we have inherited an interface that's quite difficult to use and that we'll want to improve it as muchas we can at the Python level. We can have a discussion about that.

Comment (hpohekar) [reply src/ansys/fluent/core/services/meshing_queries.py:666]: Sure.

Comment (hpohekar) [reply src/ansys/fluent/core/services/meshing_queries.py:706]: @seanpearsonuk 

We have improved error handling by introducing a function to give allowed arguments. 

Here is an example - 

image

`--------------------------------------------`

image

Comment (hpohekar) [reply src/ansys/fluent/core/services/meshing_queries.py:727]: image

Comment (hpohekar) [reply src/ansys/fluent/core/services/meshing_queries.py:727]: image

Comment (hpohekar) [reply src/ansys/fluent/core/services/meshing_queries.py:727]: image

**Comment by hpohekar** [comment]
@seanpearsonuk regarding following mesh utility function,

image

We have useful supporting  function to guide user about allowed arguments to above function.

image

`---------------------------------------`

image

**Comment by seanpearsonuk** [comment]
> @seanpearsonuk regarding following mesh utility function,
> 
> image
> 
> We have useful supporting function to guide user about allowed arguments to above function.
> 
> image
> 
> `---------------------------------------`
> 
> image

I believe that we should use enums for such things.

**Comment by hpohekar** [comment]
@seanpearsonuk 

This PR is now ready and consistent with what we are offering in 23R2. 

It contains all required meshing queries utility functions, tests and documentation.

We will implement new changes in next PR as per our  latest discussion.

Thanks.

**Review by seanpearsonuk — APPROVED** [review]
streaming doc changes in follow-up

**Review by prmukherj — APPROVED** [review]

**Pull request merged**